### DISCUSSION TITLE
Enhance documentation for properties in records

### DISCUSSION THREAD

**Discussion by MarvinKlein1508** [dotnet/csharplang#4172] [Language Ideas] [answered]
Please improve the documentation for properties of records. Consider the following record:
`public record Person(string FirstName, string LastName);`

If I want to work with IntelliSense now or provide people more information about the property then I need to specify the properties within the record again. Just like this:
```
public record Person(string FirstName, string LastName)
{
	/// <summary>
	/// Gets the firstname of Person
	/// </summary>
	public string FirstName { get; init; } = FirstName;

	/// <summary>
	/// Gets the lastname of Person
	/// </summary>
	public string LastName { get; init; } = LastName;
}
```
This is much unneccasry code for something simple as documentation. A better way would beto include the documentation directly within the documentation for the record. For example:
```
/// <summary>
/// Represents a real person
/// </summary>
/// <param val="FirstName">Gets the firstname of person</param>
/// <param val="LastName">Gets the lastname of person</param>
public record Person(string FirstName, string LastName);
```
 It is much easier to read and define and keeps the record definition really slim. You simply enter the documentation for all properties that you want to document. All other properties will have none.

What do you think?
-Marvin Klein

**Comment by CyrusNajmabadi** [comment, marked as answer]
https://github.com/dotnet/roslyn/pull/49134

**Reply by jnm2** [reply]
Discussion: https://github.com/dotnet/roslyn/issues/44571
